File: pytools/data/data.code.csharp.client/ReqSceneMove.cs
using System.Collections;
using System.Collections.Generic;


public class ReqSceneMove
{
	private MsgSceneRotPos _scene_rot_pos;
	private MsgSceneVector3 _forward;
	private string _ani_name;
	private short _x_axis;


	public Packet Encode()
	{
		Packet packet = new Packet();
		packet.WriteBuffer(this._scene_rot_pos.GetBuffer());
		packet.WriteBuffer(this._forward.GetBuffer());
		packet.WriteString(this._ani_name);
		packet.WriteShort(this._x_axis);
		packet.Encode(Msg.P_REQ_SCENE_MOVE);
		return packet;
	}


	public MsgSceneRotPos scene_rot_pos
	{
		get { return this._scene_rot_pos; }
		set { this._scene_rot_pos = value; }
	}

	public MsgSceneVector3 forward
	{
		get { return this._forward; }
		set { this._forward = value; }
	}

	public string ani_name
	{
		get { return this._ani_name; }
		set { this._ani_name = value; }
	}

	public short x_axis
	{
		get { return this._x_axis; }
		set { this._x_axis = value; }
	}

}

File: pytools/data/data.code.java.client/ReqSceneMove.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class ReqSceneMove
{
	private MsgSceneRotPos scene_rot_pos;
	private MsgSceneVector3 forward;
	private String ani_name;
	private short x_axis;


	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeBytes(scene_rot_pos.getBytes());
		packet.writeBytes(forward.getBytes());
		packet.writeString(ani_name);
		packet.writeShort(x_axis);
		return packet.encode((short)Msg.P_REQ_SCENE_MOVE);
	}



	public void setSceneRotPos(MsgSceneRotPos scene_rot_pos)
	{
		this.scene_rot_pos = scene_rot_pos;
	}

	public void setForward(MsgSceneVector3 forward)
	{
		this.forward = forward;
	}

	public void setAniName(String ani_name)
	{
		this.ani_name = ani_name;
	}

	public void setXAxis(short x_axis)
	{
		this.x_axis = x_axis;
	}

}
